openfisca_web_api/wsgihelpers.py

```diff
@@ -96,10 +96,13 @@ def respond_json(ctx, data, code = None, headers = None, jsonp = None):
                 for name, value in error.iteritems()
                 if value is not None
                 )
+        if isinstance(error, str):
+            error = data['error'] = dict(
+                message = data['error'],
+                code = 500
```

Comment:
> I don't think it is a good idea to hardcode 500, the code argument of respond_json should be used
That is the case  ;)
I'm just setting a default for `error['code']`, which is evaluated lines 111-112 as `code = code or error['code']`.

> I would prefer to add an assertion
If you have a failing assertion in this part of the code, you will crash and send a text-only `Internal Server Error` response.

This patch is not great, but it is what amounted to the least refactor of the existing heuristics in the `if error:` block.